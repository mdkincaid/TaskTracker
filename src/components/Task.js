// react
import { FaEdit, FaTimes } from 'react-icons/fa';

// third-party
import moment from 'moment';

// project
import EditTask from './EditTask';

const Task = ({ task, onDelete, onToggle, onEdit, editing, setEditing }) => {
	return (
		<div
			className={`task ${task.reminder ? 'reminder' : ''}`}
			onDoubleClick={() => onToggle(task.id)}
		>
			{editing === task.id ? (
				<div>
					<EditTask
						onEdit={onEdit}
						task={task}
						setEditing={setEditing}
					/>
				</div>
			) : (
				<div>
					{' '}
					<h3>
						{task.text}
						<FaEdit
							style={{ marginLeft: 'auto' }}
							onClick={() => setEditing(task.id)}
						/>
						<FaTimes
							style={{ color: 'red', cursor: 'pointer' }}
							onClick={() => onDelete(task.id)}
						/>
					</h3>
					<p>{moment(task.date).format('dddd MMM Do [at] h:mma')}</p>
				</div>
			)}
		</div>
	);
};

export default Task;
