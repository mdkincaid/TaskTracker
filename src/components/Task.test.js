// react
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render, screen } from '@testing-library/react'

// project
import Task from './Task'

describe('Task', () => {
    it('should render correctly', () => {
        // arrange
        const task = {
            text: 'Test Task',
            date: '2021-06-01T16:30:00.000Z',
            reminder: false,
            id: 1,
        }
        const onDelete = () => {}
        const onToggle = () => {}
        const onEdit = () => {}
        const editing = 0
        const onEditSelect = () => {}

        // act
        const { container } = render(
            <Task
                task={task}
                onDelete={onDelete}
                onToggle={onToggle}
                onEdit={onEdit}
                editing={editing}
                onEditSelect={onEditSelect}
            />
        )

        // assert
        expect(container).toMatchSnapshot()
    })

    it('should render the task text', () => {
        // arrange
        const task = {
            text: 'Test Task',
            date: '2021-06-01T16:30:00.000Z',
            reminder: false,
            id: 1,
        }
        const onDelete = () => {}
        const onToggle = () => {}
        const onEdit = () => {}
        const editing = 0
        const onEditSelect = () => {}

        // act
        render(
            <Task
                task={task}
                onDelete={onDelete}
                onToggle={onToggle}
                onEdit={onEdit}
                editing={editing}
                onEditSelect={onEditSelect}
            />
        )

        // assert
        expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent(task.text)
    })
})
