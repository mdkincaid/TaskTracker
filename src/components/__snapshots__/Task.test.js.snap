// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should render correctly 1`] = `
<div>
  <div
    class="task "
  >
    <div>
       
      <h3>
        Test Task
      </h3>
      <button
        aria-label="edit task 1"
        style="margin-left: auto;"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          />
        </svg>
      </button>
      <button
        aria-label="delete task 1"
        style="color: red; cursor: pointer;"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          />
        </svg>
      </button>
      <p>
        Tuesday Jun 1st at 12:30pm
      </p>
    </div>
  </div>
</div>
`;
